### ISSUE TITLE
[Feature]: Cannot grant access to WebUSB device, "Connect" dialog cannot be managed

### ISSUE THREAD

**Issue by busticated** [puppeteer/puppeteer#8813]
### Bug description

Steps to reproduce the problem:

1. Launch a browser: 

    ```js
    const browser = await puppeteer.launch();
    ```

2. Navigate to a url w/ WebUSB functionality:

    ```js
    const page = await browser.newPage();
    await page.goto('https://example.com/webusb-test'); // replace example url
    ```

3. Listen for "Connect" dialog and accept device:

    ```js
    page.on("dialog", async dialog => {
        await dialog.accept('my-device-name'); 
    });
    ```

4. Request a device:

    ```js
    await page.evaluate(() => {
        const device = await navigator.usb.requestDevice({ filters: [{ vendorId: <my-id> }] });
    });
    ```

Dialog launches but i'm not able to programmatically interact with it to select and approve my device. 

### Puppeteer version

16.1.1

### Node.js version

16.16.0

### npm version

8.11.0

### What operating system are you seeing the problem on?

macOS

**Comment by OrKoN** [comment]
The dialog [is for JS alert/confirm dialogs](https://chromedevtools.github.io/devtools-protocol/tot/Page/#type-DialogType) etc. I think WebUSB is using a different mechanism that is more similar to permissions than to JS-driven dialogs. Have you tried to override the dialog using browser permission overrides https://pptr.dev/api/puppeteer.browsercontext.overridepermissions? if that does not work, I think it's a feature request.

**Comment by busticated** [comment]
>Have you tried to override the dialog using browser permission overrides

no, but i'll give that a shot and share my findings here. i haven't used that API before though so i'm not sure what value(s) to try. any tips there would be much appreciated. is it at all related to `WebUsbAllowDevicesForUrls`? ([docs](https://chromeenterprise.google/policies/?policy=WebUsbAllowDevicesForUrls))

**Comment by busticated** [comment]
ok, using the following routine:

```js
async function launchBrowser(url, permissions){
    const browser = await puppeteer.launch({ headless: false, slowMo: 1000 });
    const context = browser.defaultBrowserContext();
    await context.overridePermissions(url, permissions);
    const page = await browser.newPage();
    await page.goto(url);
    return { browser, page };
}
```

....with `permissions` set to `['webusb']`, i get:

```
Error: Unknown permission: webusb
```

these values fail in the same way as well: `['usb']`, `['WebUsbAllowDevicesForUrls']` - none are recognized and the process errors out. i even tried changing the casing - e.g. `webusb` -> `WebUSB`, `WebUsb`, `WEBUSB`. [this](https://support.google.com/chrome/a/answer/7515036?hl=en) suggests maybe the value should be `'USB'` but that didn't work either.

so, i'm a bit stumped on what the correct permission key should be (or even if there is one at all) 🤷‍♂️ 

fwiw, i also found my way to: `chrome://usb-internals/` ([info](https://codereview.chromium.org/2204713006/)) but haven't figured out how to make effective use of it. i was thinking maybe this could let me pre-approve a specific device and thus bypass the "Connect" dialog entirely via `navigator.usb.getDevices()` but no luck there yet either 😭

**Comment by OrKoN** [comment]
@busticated thanks for the investigation! As I am not really sure if this can be solved via permissions, I assume the support for handling this dialog is missing from CDP.

**Comment by OrKoN** [comment]
Filed https://crbug.com/1355742

**Comment by busticated** [comment]
@OrKoN ah, makes sense. thanks for filing that. i starred it and will follow along 🙏👍
